### PR TITLE
MVD runs in local dev env

### LINKED ISSUE #191: Run MVD on Local Machine
Feature #157
After #189, #56 

## Description

A deployment to Azure that uses Azure resources like KeyVault is provided by agera-edc/MinimumViableDataspace#139, which is a great way to get started with the MVD and deploy it to a cloud environment. 

To avoid the requirement to have an Azure subscription for hackathons and to conduct hackathons in a cloud agnostic way, it is required to run the MVD on a local machine.

_Note : In story #56 a docker-compose based approach using Azurite was created to run MVD for CI/CD without cloud dependencies. One pending component to add is the MVD UI (DataDashboard) in the docker-compose setup._

## Acceptance Criteria

- [x] The MVD can be run on a local machine based on step-by-step instructions 
- [x] Keys are stored on the local file system 
- [x] Documentation contains a security warning that keys will not be securely stored, and therefore only temporary keys for development purposes shall be used 
- [x] MVD can be executed locally on major platforms e.g. Windows, Linux based systems, MacOS
- [x] Integration test in CI to validate the local machine MVD

## Out of Scope
 
- Deployment to a local cluster, e.g. Minikube
- Additional data transfers in addition to blob storage

## Tasks

- [x] Add MVD UI in docker compose based setup created under #56.
- [x] Step by step documentation to run MVD locally.
- [x] Validate that MVD can be executed locally on major platforms e.g. Windows, Linux based systems, MacOS
- [x] Validate that the local machine MVD is validated in CI

### PR DESCRIPTION
**Pull request by pkirch**
## What this PR changes/adds

- Extend docker compose to include EDC Data Dashboards.
- Extend documentation with instructions to set up a local development environment.
- Extend CI to include UI profile to spin up EDC Data Dashboards.
- Fixed references in `system-tests/README.md`.

## Why it does that

- Support the execution of MVD in a local develop environment by extending documentation and docker compose.
- Ensure docker compose file containing EDC Data Dashboards stays functional.

## Further notes

CI: Data Dashboard containers are created by Docker Compose with UI profile. [Link to successful run](https://github.com/agera-edc/MinimumViableDataspace/runs/7747809096?check_suite_focus=true#step:9:1464).

```
Container consumer-eu-datadashboard  Healthy
Container consumer-us-datadashboard  Healthy
Container provider-datadashboard  Healthy
```

## Linked Issue(s)

Closes #191 

## Checklist

- [ ] added appropriate tests?
- [ ] performed checkstyle check locally?
- [ ] added/updated copyright headers?
- [ ] documented public classes/methods?
- [ ] added/updated relevant documentation?
- [ ] added relevant details to the changelog? (_skip with label `no-changelog`_)
- [ ] formatted title correctly?

### CODE REVIEW

File: .github/actions/gradle-setup/action.yml
Comment (cpeeyush) [inline .github/actions/gradle-setup/action.yml:34]: with this we can also update Build-Datadasboard step checkout https://github.com/agera-edc/MinimumViableDataspace/blob/65c4c6b49bc6c8fca8ed875cb7a2e3b83a5d6b9e/.github/workflows/deploy.yaml#L107.

what do you think ?

Comment (pkirch) [reply .github/actions/gradle-setup/action.yml:34]: Good catch! Will remove the redundant checkout.

**Review by paullatzelsperger — APPROVED** [review]
cool stuff! Good to go once the checks are green.

File: system-tests/docker-compose.yml
Comment (marcgs) [inline system-tests/docker-compose.yml]: Namings need adaptations according to https://github.com/agera-edc/MinimumViableDataspace/pull/242

Comment (cpeeyush) [reply system-tests/docker-compose.yml]: Upstream is still open https://github.com/eclipse-dataspaceconnector/MinimumViableDataspace/pull/49. So maybe we can do it once this goes upstream.

File: README.md
Comment (marcgs) [inline README.md]:
```suggestion
In addition to the services described in [system-tests/README.md](system-tests/README.md) the local development setup
```

File: system-tests/docker-compose.yml
Comment (pkirch) [reply system-tests/docker-compose.yml]: Thanks for the heads-up @marcgs!

I would agree with @cpeeyush to wait until upstream merges. Otherwise, this PR would be inconsistent in itself.

File: system-tests/resources/appconfig/consumer-eu/app.config.json
Comment (cpeeyush) [inline system-tests/resources/appconfig/consumer-eu/app.config.json]: nit: will this link works as expected? I mean when we click this in UI, will it open and highlight the correct transferred asset.

in terraform we have something like
storageexplorer://v=1&accountid=${azurerm_resource_group.participant.id}/providers/Microsoft.Storage/storageAccounts/{{account}}&subscriptionid=${data.azurerm_subscription.current_subscription.subscription_id}&resourcetype=Azure.BlobContainer&resourcename={{container}}


Just a question,

Comment (pkirch) [reply system-tests/resources/appconfig/consumer-eu/app.config.json]: Direct links to Storage Explorer are only supported for subscription nodes, not for local accounts.

This link provides at least that the Storage Explorer opens (I've tested this). Otherwise, we would have a dead button in the MVD UI.

For reference:
https://docs.microsoft.com/en-us/azure/storage/common/storage-explorer-direct-link#direct-link-limitations

> Direct links are only supported for resources under subscription nodes. Additionally, only the following resource types are supported:

> Storage Accounts
Blob Containers
Queues
File Shares
Tables

File: README.md
Comment (marcgs) [inline README.md]: For bash you should use slashes '/' instead of backslashes

**Review by cpeeyush — APPROVED** [review]

File: README.md
Comment (pkirch) [reply README.md]: Good catch! Fixed this.

File: README.md
Comment (marcgs) [inline README.md]: I think this paragraph will be the first time the reader encounters the local development setup topic. I would opt for not assuming he or she is familiar with the information available in the system test README, and adapt the wording accordingly.

**Review by cpeeyush — COMMENTED** [review]
Added sugegstion

File: system-tests/docker-compose.yml
Comment (pkirch) [reply system-tests/docker-compose.yml]: Adapted to new naming.

**Review by cpeeyush — APPROVED** [review]
Approving based on that existing comment about doc will be addressed.

//cc @pkirch @marcgs

**Review by marcgs — APPROVED** [review]

File: system-tests/resources/appconfig/consumer-eu/app.config.json
Comment (marcgs) [reply system-tests/resources/appconfig/consumer-eu/app.config.json]: I would mention in the doc that the "Open in Storage Explorer" button does not work, can be confusing otherwise

**Review by marcgs — APPROVED** [review]

File: README.md
Comment (pkirch) [reply README.md]: I've changed the wording. Thanks for sharing your point of view.

**Pull request merged**
